### PR TITLE
fix: Removed mentions specific spoiler code

### PR DESCRIPTION
**Pull request by tankerkiller125**
Fixes: **flarum/core#2053** 

Removed mentions specific less code, also fixed links and code in spoiler.

Before Clicking:
image

After Clicking:
image

### CODE REVIEW

**Comment by tankerkiller125** [comment]
Updated the LESS to be far more universal, fixing all images, links, text, mentions, etc. Also makes it easier to add more as we find issues.

File: less/forum.less [hunk 1]
```diff
@@ -5,10 +5,15 @@
...
-    background: #444;
-    color: transparent;
+span.spoiler {
+  background: #444;
+  color: transparent;
+  a, a:link, a:visited, a:hover, a:active {
     padding: 0;
+    color: transparent;
+    cursor: default;
+  }
```
Comment (franzliedke) [inline less/forum.less:16]: Could we turn the styling for links into this?

~~~css
* {
  color: transparent;
  background: transparent;
  pointer-events: none;
}
~~~

Would that still work? That would be super generic. :grin:

Comment (franzliedke) [reply less/forum.less:16]: It's also curious that the old code had an explicit background for the link (line 10), but your new proposal does not.

**Comment by franzliedke** [comment]
Can you expand on why it works without changes for the block spoiler?

**Comment by tankerkiller125** [comment]
The block spoiler is rendered as a details tag, what this means is that at no time is the content actually "shown" until the user clicks on it. Essentially it's like a drop down menu, the contents of said menu aren't visible at all (not even under something) until you click on it. Thus you can put anything you want inside the block spoiler and it won't be shown until you click the "details" drop down button.

We could do a little work on the details header/button if we wanted for styling purposes but it does not affect the actual functionality.

**Pull request merged**